fence/agents/gce/fence_gce.py

```diff
@@ -0,0 +1,110 @@
+#!@PYTHON@ -tt
+
+import atexit
+import sys
+sys.path.append("@FENCEAGENTSLIBDIR@")
+from googleapiclient import discovery
+from oauth2client.client import GoogleCredentials
+
+from fencing import *
+from fencing import fail, fail_usage, EC_TIMED_OUT, run_delay
+
+
+def get_nodes_list(conn, options):
+	result = {}
+	try:
+		instanceList = conn.instances().list(project=options["--project"], zone=options["--zone"]).execute()
+		for instance in instanceList["items"]:
+			status = "unknown"
+			if instance["status"] == "RUNNING":
+				status = "on"
+			elif instance["status"] == "TERMINATED":
+				status = "off"
+			result[instance["id"]] = (instance["name"], status)
+	# TODO: check which Exceptions it can throw
+	except:
+		fail_usage("Failed: Unable to connect to GCE. Check your configuration.")
+
+	return result
+
+def get_power_status(conn, options):
+	try:
+		instance = conn.instances().get(project=options["--project"], zone=options["--zone"],
+						instance=options["--plug"]).execute()
+		if instance["status"] == "RUNNING":
+			return "on"
+		elif instance["status"] == "TERMINATED":
+			return "off"
+		else:
+			return "unknown"
+	# TODO: check which Exceptions it can throw
+	except:
+		fail_usage("Failed: Unable to connect to GCE. Check your configuration.")
+
+def set_power_status(conn, options):
+	try:
+		if (options["--action"]=="off"):
+			conn.instances().stop(project=options["--project"], zone=options["--zone"],
+							instance=options["--plug"]).execute()
+		elif (options["--action"]=="on"):
+			conn.instances().start(project=options["--project"], zone=options["--zone"],
+							instance=options["--plug"]).execute()
+	# TODO: check which Exceptions it can throw
+	except :
+		fail_usage("Failed: Unable to connect to GCE. Check your configuration.")
+
+def define_new_opts():
+	all_opt["zone"] = {
+		"getopt" : "z:",
```

Comment:
Can you remove the short option here? We dont use those for new parameters as they often clash with parameters used in other agents.

Reply:
Done